### PR TITLE
Added ItemManipulationModel

### PR DESCRIPTION
**Pull request by d2dyno1**
**Resolved / Related Issues**
- For #4180

**Details of Changes**
This PR introduces `ItemManipulationModel` class that contains functions to manipulate BaseLayout items.

**Motivation**
Use ItemManipulationModel instead of `IShellPage.SlimContentPage` to later get rid of that reference.

**Validation**
How did you test these changes?
- [x] Built and ran the app
- [x] Tested the changes for accessibility

### CODE REVIEW

File: Files/Views/LayoutModes/GenericFileBrowser.xaml.cs [hunk 1]
```diff
@@ -102,9 +102,139 @@ public GenericFileBrowser()
...
+            ItemManipulationModel.RefreshItemsOpacityInvoked += ItemManipulationModel_RefreshItemsOpacityInvoked;
+        }
+
+        private void ItemManipulationModel_RefreshItemsOpacityInvoked(object sender, EventArgs e)
+        {
+            foreach (ListedItem listedItem in (IEnumerable)AllView.ItemsSource)
+            {
+                if (listedItem.IsHiddenItem)
+                {
+                    listedItem.Opacity = 0.4;
```
Comment (winston-de) [inline Files/Views/LayoutModes/GenericFileBrowser.xaml.cs:126]: Should the ```DimItemOpacity``` constant be used here?

File: Files/Views/LayoutModes/ColumnViewBrowser.xaml.cs [hunk 2]
```diff
@@ -57,6 +57,119 @@ public ColumnViewBrowser() : base()
...
+            ItemManipulationModel.RefreshItemsOpacityInvoked += ItemManipulationModel_RefreshItemsOpacityInvoked;
+        }
+
+        private void ItemManipulationModel_RefreshItemsOpacityInvoked(object sender, EventArgs e)
+        {
+            foreach (ListedItem listedItem in (IEnumerable)FileList.ItemsSource)
+            {
+                if (listedItem.IsHiddenItem)
+                {
+                    listedItem.Opacity = 0.4;
```
Comment (winston-de) [inline Files/Views/LayoutModes/ColumnViewBrowser.xaml.cs:81]: Should the ```DimItemOpacity``` constant be used here as well?

Comment (d2dyno1) [reply Files/Views/LayoutModes/ColumnViewBrowser.xaml.cs]: Done!

**Review by winston-de — DISMISSED** [review]
Lgtm, great work!

**Review by winston-de — APPROVED** [review]

**Comment by d2dyno1** [comment]
@winston-de Ok, good now! Visual Studio had a seizure so I had.

**Review by winston-de — APPROVED** [review]

**Pull request merged**